src/interpreter/mod.rs

```diff
@@ -129,9 +129,9 @@ enum TerminatorTarget {
 }
 
 #[derive(Clone, Debug, Eq, PartialEq, Hash)]
-enum ConstantId {
+enum ConstantId<'tcx> {
     Promoted { index: usize },
-    Static { def_id: DefId },
+    Static { def_id: DefId, substs: &'tcx Substs<'tcx> },
```

Comment:
Can `Promoted` constants requires substs, too? For example:

```rust
struct Foo<T>(T);
fn foo() {
    let x = &Foo(42); // The Foo(42) constant could be promoted, right?
}
```

@eddyb ^ confirm/deny?